src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+
+.EXAMPLE
+New-SafeguardTag -Name "Non-prod" -Description "server environment" -Owner "Admin1","Admin2"
+
+#>
+function New-SafeguardTag {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId=$null,
+        [Parameter(Mandatory=$true, Position=0)]
+        [string]$Name,
+        [Parameter(Mandatory=$false)]
+        [string]$Description=$null,
+        [Parameter(Mandatory=$false)]
+        [PSCustomObject]$AssetTaggingRule=$null,
+        [Parameter(Mandatory=$false)]
+        [PSCustomObject]$AssetAccountTaggingRule=$null,
+        [Parameter(Mandatory=$false)]
+        [string[]]$Owner=@()
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    # use the helper function Resolve-AssetPartitionId to get the asset partition id.
+    $AssetPartitionId = Resolve-AssetPartitionId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId
+    $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags"
+
+    $local:Body = @{
+        AssetPartitionId = $AssetPartitionId;
+        Name = "$Name";
+    }
+    if ($PSBoundParameters.ContainsKey("Description")) { $local:Body.Description = $Description }
+    if ($PSBoundParameters.ContainsKey("AssetTaggingRule")) { $local:Body.AssetTaggingRule = $AssetTaggingRule }
+    if ($PSBoundParameters.ContainsKey("AssetAccountTaggingRule")) { $local:Body.AssetAccountTaggingRule = $AssetAccountTaggingRule }    
+    if ($PSBoundParameters.ContainsKey("Owner")) {
+        if (-not (Get-Module users)) {
+            Import-module "$PSScriptRoot\users.psm1"
+        }
+        $local:Body.ManagedBy = @()
+        $Owner | ForEach-Object {
+            $local:Body.ManagedBy += (Resolve-SafeguardUserObject -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure $_)
+        }
+    }
+    Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core POST "$local:RelPath" -Body $local:Body
+}
+
+
+<#
+.SYNOPSIS
+Update an existing tag via the REST API.
+
+.DESCRIPTION
+Update an existing tag. The tag is retrieved based on the ID of tag (not the name of the tag).
+This allows the Name parameter to be used to set a new name for tag.
+Please note that changing a Tag name may impact dynamic asset groups and dynamic account groups which use "Contains" in the rule.
+If an value is set on the tag in safeguard but is not passed in as a parameter, then the value will be cleared.
```

Comment:
"...on the tag in safeguard..." The word Safeguard should be capitalized, as a proper noun.

Reply:
Also, "If an value..." should be "If a value...".